core/src/main/scala/org/apache/spark/scheduler/TaskInfo.scala

```diff
@@ -82,7 +82,7 @@ class TaskInfo(
   def status: String = {
     if (running) {
       "RUNNING"
-    } else if (gettingResult) {
+    } else if (gettingResult && running) {
```

Comment:
If we get to this condition, `running` must be false right? Should it be...

```
if (running) {
  if (gettingResult) {
    ...
  } else {
    ...
  }
}
```
?